src/linear_algebra/finite_dimensional.lean

```diff
@@ -850,6 +850,61 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma findim_eq_zero_iff_subsingleton [finite_dimensional K V] :
+  findim K V = 0 ↔ subsingleton V :=
+begin
+  rw [← not_iff_not, ← not_nontrivial_iff_subsingleton, not_not],
+  change findim K V ≠ 0 ↔ _,
+  rw ← pos_iff_ne_zero,
+  exact findim_pos_iff
+end
```

Comment:
I think this belongs near line 330, just after findim_pos_iff. It would be good to add a second lemma `findim_eq_zero` which takes `subsingleton` via typeclass resolution like `findim_pos` too

Reply:
I decided to name the second lemma `findim_eq_zero_of_subsingleton` since `findim_eq_zero` already exists